src/conf.ml

```diff
@@ -27,10 +27,21 @@ let auth =
     None
   )
 
-let pool_name os_family arch =
+let pool_name (distro:Dockerfile_distro.t) arch =
+  let os_family = Dockerfile_distro.os_family_of_distro distro in
   let os_str = match os_family with
-  | `Windows | `Cygwin -> "windows"
-  | `Linux -> "linux" in
+  | `Windows | `Cygwin ->
+    let dedicated_pool = [`V1809; `V2004; `V20H2; `Ltsc2022] in
+    begin match Dockerfile_distro.resolve_alias distro with
+    | `Windows (_, release) | `Cygwin release when List.mem release dedicated_pool ->
+      (* Temporary but convenient alias until Microsoft releases 21H1 images. *)
+      let release = if release = `Ltsc2022 then `V21H1 else release in
+      "windows-" ^ Dockerfile_distro.win10_release_to_string release
```

Comment:
The special case is fine for now - it's still not totally clear to me what this will continue to look like. Server 2022 is _based_ on 21H1 but isn't actually 21H1. These tags aren't particularly exposed to users, so I don't think we should worry too much at this stage.